lib/pdk/tests/unit.rb

```diff
@@ -21,6 +23,26 @@ def self.invoke(tests, report = nil)
         output = PDK::CLI::Exec.execute(cmd(tests))
         report.write(output) if report
       end
+
+      def self.list
+        command_argv = ['rspec', '--dry-run', '-fjson']
+        list_command = PDK::CLI::Exec::Command.new(*command_argv)
+        list_command.context = :module
+        output = list_command.execute!
+        rspec_json_output = JSON.parse(output[:stdout])
+        if rspec_json_output['examples'].empty?
+          rspec_message = rspec_json_output['messages'][0]
+          raise PDK::CLI::FatalError, _('Unable to enumerate examples. rspec reported: %{message}' % { message: rspec_message }) unless rspec_message == 'No examples found.'
```

Comment:
I'm not a fan of the 'guard clause' that rubocop pushes me towards here - the 'if' clause of this raise is off the edge of the screen :/